### ISSUE TITLE
missing-playwright-await does not work properly inside a ternary

### ISSUE THREAD

**Issue by thomas-phillips-nz** [playwright-community/eslint-plugin-playwright#204]
Hi

I encountered the missing-playwright-await rule incorrectly flagging calls to `expect` inside a ternary condition with an outer `await`:

```
const expectedCheckedStatus = true;
await (expectedCheckedStatus
      ? expect(page.locator('input')).toBeChecked()
      : expect(page.locator('input')).not.toBeChecked());
```

This incorrectly flags the `expect` statement as needing to be awaited

**Comment by mskelton** [comment]
Using the `checked` option is the correct way to handle this situation.

```js
await expect(page.locator('input')).toBeChecked({ checked: expectedCheckedStatus })
```

**Comment by thomas-phillips-nz** [comment]
True, but I think this does not address the more general issue of the expect being incorrectly flagged inside a ternary, e.g.:

```
async function testing(page: Page, commands: string) {
  const expectPromise =
    commands === ''
      ? expect(page.getByText('name')).toBeVisible()
      : expect(page.getByText('otherName')).toBeChecked();

  await expectPromise;
}
```

I know there might a better way to write this, but there still appears to be a problem with incorrectly flagging the `expect` statements inside a ternary.

**Comment by thomas-phillips-nz** [comment]
Oops sorry I edited the original comment to have a different example.

**Comment by mskelton** [comment]
@thomas-phillips-nz Not sure what you mean, both the examples you've provided in this issue I provided alternatives for.
